crates/revm/src/evm_impl.rs

```diff
@@ -91,6 +91,99 @@ impl<'a, GSPEC: Spec, DB: Database, const INSPECT: bool> EVMImpl<'a, GSPEC, DB,
         }
         Ok(())
     }
+
+    /// Set gas to the gas limit and spend it all.
+    #[cfg(not(feature = "optimism"))]
+    #[inline]
+    fn use_gas(
+        &mut self,
+        gas_limit: u64,
+        ret_gas: Gas,
+        exit_reason: InstructionResult,
+    ) -> Result<Gas, EVMError<DB::Error>> {
+        let mut gas = Gas::new(gas_limit);
+        gas.record_cost(gas_limit);
```

Comment:
I think that these 2 lines can be pulled back up into the other fn, is the same in both impls.

Reply:
That SGTM